From b474db7c3809a74a6aacb3c687cf32632c0aaa33 Mon Sep 17 00:00:00 2001
From: Marie-Cecile Afantenos <marie.cecile.afantenos@intel.com>
Date: Wed, 26 Jun 2013 19:18:32 +0200
Subject: [PORT FROM MAIN]add board id indication

BZ: 116281

This patch is a merge of the patches:
http://android.intel.com:8080/#/c/82217/
http://android.intel.com:8080/#/c/76551

Change-Id: I4d5d2dbe92abc782a12dca991994f2bcf156de61
Orig-Change-Id: If792dc814c0b7697f6f6f4f12d1626dd863056b5
Category: aosp improvement
Domain: PUPDR-Flashing
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Marie-Cecile Afantenos <marie.cecile.afantenos@intel.com>
---
 init/init.c |   30 ++++++++++++++++++++++++++++++
 1 files changed, 30 insertions(+), 0 deletions(-)

diff --git a/init/init.c b/init/init.c
index 53be596..14f6c9c 100644
--- a/init/init.c
+++ b/init/init.c
@@ -782,6 +782,34 @@ static void import_kernel_nv(char *name, int for_emulator)
     }
 }
 
+static void boardid_init (void)
+{
+    int fd = open("/proc/boardid", O_RDONLY);
+    char buf[PROP_VALUE_MAX]={'\0'};
+    char bid[PROP_VALUE_MAX]={'\0'};
+
+    if (fd >= 0) {
+        int n=read(fd, buf, PROP_VALUE_MAX-1);
+        close(fd);
+
+        if (n <= 0){
+            ERROR("fail to read /proc/boardid!\n");
+            return;
+        }
+    } else {
+        ERROR("fail to open /proc/boardid!\n");
+        return;
+    }
+
+    sscanf(buf, "boardid=%91s", bid);
+
+    if (bid[0] == '\0') {
+        ERROR("no bid value in /proc/boardid!\n");
+    }
+
+    property_set("ro.board.id", bid);
+
+}
 static void export_kernel_boot_props(void)
 {
     char tmp[PROP_VALUE_MAX];
@@ -831,6 +859,8 @@ static void export_kernel_boot_props(void)
         property_set("ro.factorytest", "2");
     else
         property_set("ro.factorytest", "0");
+
+    boardid_init();
 }
 
 static void process_kernel_cmdline(void)
-- 
1.7.4.1

